src/Microsoft.DotNet.Tools.Publish/PublishCommand.cs

```diff
@@ -236,5 +186,58 @@ private static void PublishFiles(IEnumerable<LibraryAsset> files, string outputP
                 File.Copy(file.ResolvedPath, Path.Combine(outputPath, Path.GetFileName(file.ResolvedPath)), overwrite: true);
             }
         }
+
+        public static int Main(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            var app = new CommandLineApplication();
+            app.Name = "dotnet publish";
+            app.FullName = ".NET Publisher";
+            app.Description = "Publisher for the .NET Platform";
+            app.HelpOption("-h|--help");
+
+            var framework = app.Option("-f|--framework <FRAMEWORK>", "Target framework to compile for", CommandOptionType.SingleValue);
+            var runtime = app.Option("-r|--runtime <RUNTIME_IDENTIFIER>", "Target runtime to publish for", CommandOptionType.SingleValue);
+            var output = app.Option("-o|--output <OUTPUT_PATH>", "Path in which to publish the app", CommandOptionType.SingleValue);
+            var configuration = app.Option("-c|--configuration <CONFIGURATION>", "Configuration under which to build", CommandOptionType.SingleValue);
+            var projectPath = app.Argument("<PROJECT>", "The project to publish, defaults to the current directory. Can be a path to a project.json or a project directory");
+
+            app.OnExecute(() =>
+            {
+                var publish = new PublishCommand();
+
+                publish.Framework = framework.Value();
+                publish.OutputPath = output.Value();
+
+                // TODO: Remove default once xplat publish is enabled.
+                publish.Runtime = runtime.Value() ?? RuntimeIdentifier.Current;
+
+                publish.ProjectPath = projectPath.Value;
+                if (string.IsNullOrEmpty(publish.ProjectPath))
+                {
+                    publish.ProjectPath = Directory.GetCurrentDirectory();
+                }
+
+                if (!publish.TryPrepareForPublish())
+                {
+                    return 1;
+                }
+
+                publish.PublishAllProjects();
+                return publish.NumberOfFailedToPublishProjects;
```

Comment:
I actually like printing summary and return 1 instead of what is happening right now. @Sridhar-MS - any counter arguments?